custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -222,18 +222,13 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
         return locationsService.getLocations(level, locationsCache, vm.selectedLocations, vm.disallowNational());
     };
 
-    var selectedLocationIndex = function () {
-        return _.findLastIndex(vm.selectedLocations, function (locationId) {
-            return locationId && locationId !== ALL_OPTION.location_id;
-        });
-    };
-
     vm.disabled = function (level) {
         return locationsService.isLocationDisabled(level, vm);
     };
 
     vm.onSelectForISSNIP = function ($item, level) {
-        var selectedLocationId = vm.selectedLocations[selectedLocationIndex()];
+        var selectedLocIndex = locationsService.selectedLocationIndex(vm.selectedLocations);
+        var selectedLocationId = vm.selectedLocations[selectedLocIndex];
```

Comment:
This isn't really an ID, its a location object

Reply:
which `vm.selectedLocations[selectedLocIndex]`?. It returns an Id only. selectedLocations is a list

Reply:
um. may be I did not follow. But `vm.selectedLocations[selectedLocIndex]` returns an Id only. as `selectedLocations` is an array.
M I missing something?